assets/js/SortFromTable.js

```diff
@@ -0,0 +1,120 @@
+            next = 0;
+        } else {
+            next = 1;
+        }
+    }
+    return next;
+}
+
+// Wrapper function for changing sort direction, since it needs to be used twice.
+function updateSortDirection(wfm, sortInfo, sortType, reset=false) {
+
+    // I left this code in because the implementation below can
+    // cause the UI to show the wrong arrow for a second before
+    // the database fully updates. We can talk about which behavior
+    // is preferable later.
+    /*
+    var direction = 0; // Corresponds to "select"
+    if(sortInfo.direction == 'ASC') {
+        direction = 1;
+    } else if(sortInfo.direction == 'DESC') {
+        direction = 2
+    }
+    */
+
+    // Sort directions: 0 -> NOP, 1 -> Ascending, 2 -> Descending
+    var directionParam = findParamValByIdName(wfm, "direction");
+    let currentDirection = reset ? 0 : parseInt(directionParam.value);
+    var nextDirection = getNextSortDirection(currentDirection, sortType)
+
+    api.onParamChanged(directionParam.id, {value: nextDirection});
+}
+
+function updateSortModule(wfm, sortInfo, sortType) {
+    var column = sortInfo.column;
+    var columnParam = findParamValByIdName(wfm, "column");
+    var typeParam = findParamValByIdName(wfm, "dtype");
+    // If column changes then we need to change both sort column and type
+    if(columnParam.value != column) {
+        api.onParamChanged(columnParam.id, {value: column})
+            .then(() => {
+                api.onParamChanged(typeParam.id, {value: sortType}).then(() => {
+                    // Timeout to prevent database locked 500 errors.
+                    setTimeout(() => {
+                        updateSortDirection(wfm, sortInfo, sortType, true);
+                    }, 200);
+                });
+            });
+    } else {
+        updateSortDirection(wfm, sortInfo, sortType);
+    }
+}
+
+export function updateSort(wfModuleId, sortInfo) {
+    var state = store.getState();
+
+    api.getColumns(wfModuleId)
+        .then((columns) => {
+            let columnInfo = columns.filter((col) => (col.name == sortInfo.column));
+            if(columnInfo.length < 1) {
+                return;
+            }
+
+            // Must be kept in sync with sortfromtable.json
+            let sortTypes = "String|Number|Date".split("|")
+            let sortType = columnInfo[0].type;
+            let sortTypeIdx = sortTypes.indexOf(sortType);
+            var existingSortModule = findModuleWithIdAndIdName(state, wfModuleId, 'sort-from-table')
+            if(existingSortModule) {
+                updateSortModule(existingSortModule, sortInfo, sortTypeIdx);
+                if(existingSortModule.id != wfModuleId) {
+                    store.dispatch(setSelectedWfModuleAction(existingSortModule.id));
+                }
+            } else {
+                let wfModuleIdx = getWfModuleIndexfromId(state, wfModuleId);
+                api.addModule(getPageID(), state.sortModuleId, wfModuleIdx + 1)
```

Comment:
This could probably be replaced with an add module action, which also sets the selected module